doc/security/sensor-threat.rst

```diff
@@ -0,0 +1,327 @@
+
+Assets
+======
+
+One aspect of the threat model to consider are assets involved in the
+operation of the device. The following list enumerates the assets
+included in this model:
+
+1. **The bootloader**. This is a small code/data image contained in
+   on-device flash that is the first code to run. In order to establish
+   a root of trust, this image must be immutable. This model assumes
+   that the SoC provides a mechanism to protect a region of the flash
+   from future writes, and that this will be done after this image is
+   programmed into the device, early in production.
+
+2. **The application firmware image**. This asset consists of the
+   remainder of the firmware run by the microcontroller. The distinction
+   is made because this part of the image will need to be updated
+   periodically as security vulnerabilities are discovered. Requirements
+   for updates to this image are:
+
+   a. The image shall only be replaced with an authorized image.
+
+   b. When an authorized replacement image is available, the update
+      shall be done in a timely manner.
+
+   c. The image update shall be seen as atomic, meaning that when the
+      image is run, the flash shall contain either the update image in
+      its entirety, or the old image in its entirety.
+
+3. **Root certificate list**. In order to authenticate the cloud service
+   (server), the IoT device must have a list of root certificates that
+   are allowed to sign the certificate on the server. For cloud-provider
+   based services, this list will generally be provided by the service
+   provider. Because the root certificates can expire, and possibly be
+   revoked, this list will need to be periodically be updated.
+
+4. **Client secrets**. To authenticate the client to the service, the
+   client must posess some kind of secret. This is general a private
+   key, usually either an RSA key or an EC private key. When
+   establishing communication with the server, the device will use
+   this secret either as part of the TLS establishment, or to sign a
+   message used in the communication.
+
+   This secret is generally generated by the service provider, or by
+   software running elsewhere, and must be securely installed on the
+   device. Policy may dictate that this secret be replaced
+   periodically, which will require a way to update the client secret.
+   Typically, the service will allow two or three active keys to allow
+   this update to proceed while the old key is used.
+
+5. **Current date/time**. TLS certificate verification requires
+   knowledge of the current date and time in order to determine if the
+   current time falls within the certificate’s current validity time.
+   Also, token based client authentication will generally require the
+   client to sign a message containing a time window that the token is
+   valid. Certificate validation requires the device notion of date and
+   time to be accurate within a day or so. Token generation generally
```

Comment:
> to be accurate within a day or so

It might be worth mentioning how this, i.e. secure timing, could be ensured

Reply:
I'll put in a paragraph explaining some of this.